test/hack/cleanup/main.go

```diff
@@ -427,13 +441,25 @@ func (ip *instanceProfile) Cleanup(ctx context.Context, names []string) ([]strin
 	return deleted, errs
 }
 
-func fireMetric(ctx context.Context, cloudWatchClient *cloudwatch.Client, name string, value float64) error {
-	_, err := cloudWatchClient.PutMetricData(ctx, &cloudwatch.PutMetricDataInput{
-		Namespace: lo.ToPtr(karpenterMetricNamespace),
-		MetricData: []cloudwatchtypes.MetricDatum{
+type timestream struct {
+	timestreamClient *timestreamwrite.Client
+}
+
+func (t *timestream) FireMetric(ctx context.Context, name string, value float64) error {
+	_, err := t.timestreamClient.WriteRecords(ctx, &timestreamwrite.WriteRecordsInput{
+		DatabaseName: aws.String(karpenterMetricDatabase),
+		TableName:    aws.String(karpenterMetricTableName),
+		Records: []timestreamtypes.Record{
 			{
-				MetricName: lo.ToPtr(name),
-				Value:      lo.ToPtr(value),
+				MeasureName:  aws.String(name),
+				MeasureValue: aws.String(fmt.Sprintf("%f", value)),
+				Time:         aws.String(fmt.Sprintf("%d", time.Now().UnixMilli())),
+				Dimensions: []timestreamtypes.Dimension{
+					{
+						Name:  aws.String("region"),
```

Comment:
What's the reason that region is a dimension?

Reply:
Since we now using resource from three region, I think we should know which region we are cleaning up the most resources

Reply:
Agreed, but this region will always be the same region, right? Since we aren't reading this from the environment? We're just statically setting the value as a constant at the top of the file?

Reply:
No, this region is pulled from the API default config.

Reply:
Updated the regions, such that the region is pulled from the config